### DISCUSSION TITLE
Android app detects that it runs on a Virtual Machine when Magisk installed.

### DISCUSSION THREAD

**Discussion by chadify** [LSPosed/MagiskOnWSALocal#679] [Q&A]
So, I'm running WSA(Windows Subsystem for Android) on my Windows 11. I've got this educational app running just fine on the stock build of WSA. The stock build is very minimal with the only other app the Amazon App store.

I wanted to root WSA, so I installed the the build with Magisk.
And now I'm getting the following error: "Please run on a Physical Device"

![image](https://github.com/LSPosed/MagiskOnWSALocal/assets/72332949/5db434a5-6cb8-493b-865d-0985b97049a9)

What dependency/framework crept up with this Magisk build such that this app started detecting the VM? Any ideas?

**Comment by rushiranpise** [comment]
does it works on normal wsa? ie without root?

**Reply by chadify** [reply]
Yes. BTW I fixed the issue by rooting with kernelISU.
